modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionFactory.java

```diff
@@ -202,25 +196,32 @@ public EndpointDefinition createDefinition(OMElement elem) {
                 SynapseConstants.SYNAPSE_NAMESPACE,
                 XMLConfigConstants.ERROR_CODES));
             if (timeoutCodes != null && timeoutCodes.getText() != null) {
-                StringTokenizer st = new StringTokenizer(timeoutCodes.getText().trim(), ", ");
-                while (st.hasMoreTokens()) {
-                    String s = st.nextToken();
-                    try {
-                        definition.addTimeoutErrorCode(Integer.parseInt(s));
-                    } catch (NumberFormatException e) {
-                        handleException("The timeout error codes should be specified " +
-                            "as valid numbers separated by commas : " + timeoutCodes.getText(), e);
+                Value timeoutErrorCodesValue = valueFactory.createTextValue(timeoutCodes);
+                if (timeoutErrorCodesValue.getKeyValue() != null) {
+                    StringTokenizer st = new StringTokenizer(timeoutCodes.getText().trim(), ", ");
```

Comment:
Any reason for splitting based on ", "? Is it possible to give comma separated values without a space?

Reply:
updated